iron-overlay-behavior.d.ts

```diff
@@ -0,0 +1,263 @@
+/// <reference path="iron-overlay-manager.d.ts" />
+/// <reference path="iron-scroll-manager.d.ts" />
+/// <reference path="iron-focusables-helper.d.ts" />
+
+declare namespace Polymer {
+
+  interface IronOverlayBehaviorImpl {
+
+    /**
+     * True if the overlay is currently displayed.
+     */
+    opened: boolean|null|undefined;
+
+    /**
+     * True if the overlay was canceled when it was last closed.
+     */
+    readonly canceled: boolean|null|undefined;
+
+    /**
+     * Set to true to display a backdrop behind the overlay. It traps the focus
+     * within the light DOM of the overlay.
+     */
+    withBackdrop: boolean|null|undefined;
+
+    /**
+     * Set to true to disable auto-focusing the overlay or child nodes with
+     * the `autofocus` attribute` when the overlay is opened.
+     */
+    noAutoFocus: boolean|null|undefined;
+
+    /**
+     * Set to true to disable canceling the overlay with the ESC key.
+     */
+    noCancelOnEscKey: boolean|null|undefined;
+
+    /**
+     * Set to true to disable canceling the overlay by clicking outside it.
+     */
+    noCancelOnOutsideClick: boolean|null|undefined;
+
+    /**
+     * Contains the reason(s) this overlay was last closed (see `iron-overlay-closed`).
+     * `IronOverlayBehavior` provides the `canceled` reason; implementers of the
+     * behavior can provide other reasons in addition to `canceled`.
+     */
+    closingReason: object|null|undefined;
+
+    /**
+     * Set to true to enable restoring of focus when overlay is closed.
+     */
+    restoreFocusOnClose: boolean|null|undefined;
+
+    /**
+     * Set to true to allow clicks to go through overlays. 
+     * When the user clicks outside this overlay, the click may
+     * close the overlay below.
+     */
+    allowClickThrough: boolean|null|undefined;
+
+    /**
+     * Set to true to keep overlay always on top.
+     */
+    alwaysOnTop: boolean|null|undefined;
+
+    /**
+     * Determines which action to perform when scroll outside an opened overlay happens.
+     * Possible values:
+     * lock - blocks scrolling from happening,
+     * refit - computes the new position on the overlay
+     * cancel - causes the overlay to close
+     */
+    scrollAction: string|null|undefined;
+
+    /**
+     * The node being focused.
+     */
+    _focusedChild: Node|null;
+
+    /**
+     * The backdrop element.
+     */
+    backdropElement(): any;
+
+    /**
+     * Returns the node to give focus to.
+     */
+    _focusNode(): any;
+
+    /**
+     * Array of nodes that can receive focus (overlay included), ordered by `tabindex`.
+     * This is used to retrieve which is the first and last focusable nodes in order
+     * to wrap the focus for overlays `with-backdrop`.
+     *
+     * If you know what is your content (specifically the first and last focusable children),
+     * you can override this method to return only `[firstFocusable, lastFocusable];`
+     */
+    _focusableNodes(): any;
```

Comment:
type `Array<Node>`
https://github.com/PolymerElements/iron-overlay-behavior/blob/master/iron-overlay-behavior.html#L178

Reply:
Done. Actually it should be `!Array<!Node>` or else it would be a nullable array of nullable nodes, since non-primitive types are nullable by default in Closure. Updated the other cases I found too.